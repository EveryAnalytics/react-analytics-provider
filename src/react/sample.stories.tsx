import React from 'react';
import {Analytics} from '../mixin';

Analytics.init({
  preset: {
    googleAnalytics: {
      trackingId: '123',
      persistentValues: {userNo: 1},
    },
    amplitude: {
      apiKey: '456',
    },
  },
  onEvent: (...args) => {
    console.info('event occured! ' + args);
  },
});

const Sample = () => {
  return (
    <button
      onClick={() => {
        Analytics.event('storybook click');
      }}
    >
      click me
    </button>
  );
};

export default {
  title: 'Sample',
  component: Sample,
};

export const Default = () => <Sample />;
