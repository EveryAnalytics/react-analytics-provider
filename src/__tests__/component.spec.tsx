import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SampleComponent from '../components/Sample';

test('component testing', () => {
  const component = renderer.create(<SampleComponent text="World" />);
  const testInstance = component.root;

  expect(testInstance.findByType(SampleComponent).props.text).toBe('World');

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
