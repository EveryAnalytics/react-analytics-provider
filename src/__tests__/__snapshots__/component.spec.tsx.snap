// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component testing 1`] = `
Array [
  <div>
    Ready to use React Hook!
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "black",
        "color": "white",
      }
    }
  >
    World
  </div>,
]
`;
